src/lib/ndpi_utils.c

```diff
@@ -2177,17 +2184,41 @@ ndpi_http_method ndpi_http_str2method(const char* method, u_int16_t method_len)
   case 'G': return(NDPI_HTTP_METHOD_GET);
   case 'H': return(NDPI_HTTP_METHOD_HEAD);
 
+  case 'L':
+    if (strncmp(method, "LOCK", 4) == 0)
+      return(NDPI_HTTP_METHOD_LOCK);
+    break;
+
+  case 'M':
+    if (strncmp(method, "MOVE", 4) == 0)
+      return(NDPI_HTTP_METHOD_MOVE);
+    else if (strncmp(method, "MKCO", 4) == 0)
+      return(NDPI_HTTP_METHOD_MKCOL);
+    break;
+
   case 'P':
     switch(method[1]) {
     case 'A':return(NDPI_HTTP_METHOD_PATCH);
     case 'O':return(NDPI_HTTP_METHOD_POST);
     case 'U':return(NDPI_HTTP_METHOD_PUT);
+    case 'R':
+      if (method_len >= 5) {
+        if (strncmp(method, "PROPF", 5) == 0)
+          return(NDPI_HTTP_METHOD_PROPFIND);
+        else if (strncmp(method, "PROPP", 5) == 0)
+          return NDPI_HTTP_METHOD_PROPPATCH;
+      }
```

Comment:
Well, it works, but doesn't seem optimal to me. @IvanNardi what do you think? Is there any way this could be improved?

Reply:
I think that part it is fine as is.
The general goal is to have "consistent/coherent" code